lpvartypes_array.pas

```diff
@@ -759,6 +763,46 @@ function TLapeType_DynArray.Eval(Op: EOperator; var Dest: TResVar; ALeft, ARight
     if wasConstant then
       Result.Writeable := False;
   end
+  else if (op = op_Plus) and (BaseType = ltDynArray) and CompatibleWith(ARight.VarType) then
+  begin 
+    {!FIXME:
+      when `Result := StaticArray + DynArray` result-type is static array, causing this to fail.
+      On the other hand I am unsure why (left) StaticArray enters here.
+      - slacky
+    }
+    Result := NullResVar;
+    Result.VarType := Self;
+    FCompiler.getDestVar(Dest, Result, op);
+
+    if (Result.VarPos.MemPos = mpStack) then
+    begin
+      tmpVar := FCompiler.getTempVar(Self);
+      Result := _ResVar.New(tmpVar);
+    end;
+
+    wasConstant := not Result.Writeable;
+    if wasConstant then Result.Writeable := True;
+
+    Result := Eval(op_Assign, tmpResVar, Result, ALeft, [], Offset, Pos);
+
+    IndexVar := _ResVar.New(FCompiler.getTempVar(ltInt32));
```

Comment:
Currently, it does not store the length here; it directly passes the length expression as a parameter, which is how it should be done :)
IdExpr and IndexVar are not used for anything useful, other than as "base nodes" for `TLapeTree_*.Create` calls.